contrib/farsh-cmake/CMakeLists.txt

```diff
@@ -0,0 +1,17 @@
+set(LIBRARY_DIR "${ClickHouse_SOURCE_DIR}/contrib/farsh")
+
+set(SRCS
+        "${LIBRARY_DIR}/farsh.h"
+        "${LIBRARY_DIR}/farsh.c"
+        )
+
+add_library(farsh ${SRCS})
+target_include_directories (farsh PUBLIC ${LIBRARY_DIR})
+
+if(HAVE_AVX2)
```

Comment:
This is wrong.

We should compile multiple versions of the function with different instruction sets and do dynamic dispatch.

I've checked the source code of FARSH and found that it does not have dynamic dispatch.

The code is simple enough, and it is ok to copy-paste it entirely for the purpose of this change.

CC @Bulat-Ziganshin

Reply:
Original code unfortunately is very specific about instruction set, so doesn't fit well. If we really need this hash we can try @Bulat-Ziganshin suggestion:
https://github.com/Cyan4973/xxHash

Reply:
We already use xxHash.